CMakeLists.txt

```diff
@@ -143,71 +113,59 @@ add_library(jluna SHARED
     include/gc_sentinel.hpp
 )
 
-set_target_properties(jluna PROPERTIES
-    LINKER_LANGUAGE C
-    LIBRARY_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR}
-)
-target_compile_features(jluna PUBLIC cxx_std_20)
-
-### C ADAPTER ##
-
-add_library(jluna_c_adapter SHARED
-    .src/c_adapter.hpp
-    .src/c_adapter.cpp
-)
-
-target_include_directories(jluna_c_adapter
-    PUBLIC $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}>
-    PUBLIC $<BUILD_INTERFACE:${JULIA_INCLUDE}>
-)
-
-target_link_libraries(jluna_c_adapter PUBLIC $<BUILD_INTERFACE:${JULIA_LIB}>)
-
-set_target_properties(jluna_c_adapter PROPERTIES
-    LINKER_LANGUAGE C
-    LIBRARY_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR}
-)
-
 target_link_libraries(jluna PUBLIC jluna_c_adapter)
+target_include_directories(jluna PRIVATE "$<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}>")
 
-### EXECUTABLES ###
-
-add_executable(JLUNA_TEST
-    .test/main.cpp
-    .test/test.hpp
-)
-target_link_libraries(JLUNA_TEST jluna ${JULIA_LIB})
+### HACK: export all symbols on Windows ###
 
-add_executable(JLUNA_BENCHMARK
-    .benchmark/main.cpp
-    .benchmark/benchmark.hpp
-    .benchmark/benchmark_aux.hpp
+set_target_properties(jluna jluna_c_adapter PROPERTIES
+    WINDOWS_EXPORT_ALL_SYMBOLS YES
+    LIBRARY_OUTPUT_DIRECTORY ${C_ADAPTER_PATH}
```

Comment:
Unquoted variable usage, but also don't hardcode `LIBRARY_OUTPUT_DIRECTORY`. Move it to a [preset](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html). You can find many examples for preset use here: https://github.com/friendlyanon/cmake-init/wiki/Examples